main.go

```diff
@@ -126,73 +109,266 @@ func (e *periodicExporter) Collect(ch chan<- prometheus.Metric) {
+
+	e.counter(metricsChan, labels, labelValues, "mesos_recovery_errors", float64(stats.RecoveryErrors), "Recovery errors")
+	e.gauge(metricsChan, labels, labelValues, "mesos_registered", float64(stats.Registered), "Registered")
+
+	e.counter(metricsChan, labels, labelValues, "mesos_tasks_failed", float64(stats.TasksFailed), "Failed tasks")
+	e.counter(metricsChan, labels, labelValues, "mesos_tasks_finished", float64(stats.TasksFinished), "Finished tasks")
+	e.counter(metricsChan, labels, labelValues, "mesos_tasks_killed", float64(stats.TasksKilled), "Killed tasks")
+	e.counter(metricsChan, labels, labelValues, "mesos_tasks_lost", float64(stats.TasksLost), "Lost tasks")
+	e.gauge(metricsChan, labels, labelValues, "mesos_tasks_running", float64(stats.TasksRunning), "Running tasks")
+	e.gauge(metricsChan, labels, labelValues, "mesos_tasks_staging", float64(stats.TasksStaging), "Staging tasks")
+	e.gauge(metricsChan, labels, labelValues, "mesos_tasks_starting", float64(stats.TasksStarting), "Starting tasks")
+
+	e.counter(metricsChan, labels, labelValues, "mesos_uptime_secs", stats.UptimeSecs, "Uptime in seconds")
+
+	e.counter(metricsChan, labels, labelValues, "mesos_valid_framework_messages", float64(stats.ValidFrameworkMessages), "Valid framework messages")
+	e.counter(metricsChan, labels, labelValues, "mesos_valid_status_updates", float64(stats.ValidStatusUpdates), "Valid status updates")
+
+	e.gauge(metricsChan, labels, labelValues, "mesos_system_cpus_total", float64(stats.SystemCpuTotal), "System CPU total")
+	e.gauge(metricsChan, labels, labelValues, "mesos_system_load_15min", stats.SystemLoad15MinuteAvg, "System load 15 minutes average")
+	e.gauge(metricsChan, labels, labelValues, "mesos_system_load_1min", stats.SystemLoad1MinuteAvg, "System load 1 minute average")
+	e.gauge(metricsChan, labels, labelValues, "mesos_system_load_5min", stats.SystemLoad5MinuteAvg, "System load 5 minutes average")
+	e.gauge(metricsChan, labels, labelValues, "mesos_system_mem_free_bytes", float64(stats.SystemMemoryFreeBytes), "System memory free in bytes")
+	e.gauge(metricsChan, labels, labelValues, "mesos_system_mem_total_bytes", float64(stats.SystemMemoryTotalBytes), "System memory total in bytes")
+}
+
+func (e *periodicExporter) fetchMasterMetricsSnapshot(url *url.URL, host string, metricsChan chan <- prometheus.Metric) {
+	monitorURL := fmt.Sprintf("%s/metrics/snapshot", url)
+	resp, err := httpClient.Get(monitorURL)
+	if err != nil {
+		log.Warn(err)
+		e.errors.WithLabelValues(host).Inc()
+		return
+	}
+	defer resp.Body.Close()
+
+	var stats mesos_metrics.MasterMetrics
+	if err = json.NewDecoder(resp.Body).Decode(&stats); err != nil {
+		log.Warn("failed to deserialize response: ", err)
+		e.errors.WithLabelValues(host).Inc()
+		return
+	}
+	labels := []string{"node", "type"}
+	labelValues := []string{host, "master"}
+	e.gauge(metricsChan, labels, labelValues, "mesos_cpus_percent", stats.CpuPercentage, "CPU percentage")
```

Comment:
aren't the labels for that? Next to first timer with Go, I also just started with Prometheus :smile:

So do I understand it correctly, that:
- Because master and slave nodes have a different purpose, their metric keys should be different and eg. contain "_master_" and "_slave_" respectively?
- The labels can be used to seperate metrics for different instances or call types?
  eg. for this situation having a "node" label with the hostname as value works nicely, because this indicates for which instance the metric is.
  And for something like `http_api_error_count` could have a label based on the specific http statuscode the error happened in?

Do I understand that correctly?